libraries/AP_Compass/AP_Compass_HMC5843.cpp

```diff
@@ -282,12 +303,11 @@ void AP_Compass_HMC5843::read()
     publish_filtered_field(field, _compass_instance);
 }
 
-
-bool AP_Compass_HMC5843::_exit_test_mode()
+bool AP_Compass_HMC5843::_setup_mode()
```

Comment:
I get it now. Well, I think the function name is ambiguous. For me it gave me the idea it was going to enter in some sort of *setup mode*, whereas what the code does is quite different. Maybe rename it to `_sampling_mode` or `_setup_sampling_mode`?